### PR TITLE
Edit an item endpoint/branch

### PR DESCRIPTION
**Pull request by KMPWeiland**
**Edit an Item Endpoint**
This PR allows users to update an item's attributes (name, description, unit_price, and merchant_id). It includes error handling for invalid item IDs and failed updates due to validation errors.

Features:
>PATCH /api/v1/items/:id endpoint to update an item.
>Returns 404 if the item is not found.
>Returns 422 for validation errors.
>RSpec test for successful updates (currently failing due to unresolved issues).

Pending: I still need to debug failing RSpec test (likely related to missing merchant data) and ddd tests for edge cases like invalid merchant_id or missing parameters. These will be included in a future branch.

### CODE REVIEW

File: spec/items_controller_spec.rb
Comment (pshim17) [inline spec/items_controller_spec.rb:1]: Good job writing the test for the update method!

File: app/models/item.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
 class Item < ApplicationRecord
-  belongs_to :merchants
+  belongs_to :merchant
```
Comment (pshim17) [inline app/models/item.rb]: Good catch here!!

File: app/controllers/api/v1/items_controller.rb [hunk 2]
```diff
@@ -1,5 +1,33 @@
 class Api::V1::ItemsController < ApplicationController
-  def index
+   
+  def update
```
Comment (pshim17) [inline app/controllers/api/v1/items_controller.rb:66]: This method looks great! I would have approached it the same way! We can work on adding a serializer to refactor and make the code look cleaner later!

**Review by pshim17 — APPROVED** [review]

File: app/controllers/api/v1/items_controller.rb (hunk 2 above)
Comment (kevin-newland) [reply app/controllers/api/v1/items_controller.rb:66]: I agree great job! I like the idea of the serializer to refactor later.

**Review by kevin-newland — APPROVED** [review]
Awesome work on the update method!

**Review by kevin-newland — APPROVED** [review]
Awesome work on the update method!
